app/src/main/contrib/components/Responsibilities/Expectations/ExpectationsDescription.tsx

```diff
@@ -1,15 +1,68 @@
 import * as React from "react";
-import {ExpectationMapping} from "../../../../shared/models/Generated";
+import {Country, ExpectationMapping, NumberRange} from "../../../../shared/models/Generated";
+import {Popover, PopoverBody} from "reactstrap";
+
+interface State {
+    showCountries: boolean;
+}
+
+interface Props {
+    countries: Country[]
+    targetKey: string;
+}
+
+export class CountriesList extends React.Component<Props, State> {
+
+    constructor(props: Props) {
+        super(props);
+
+        this.toggle = this.toggle.bind(this);
+        this.state = {
+            showCountries: false
+        };
+    }
+
+    toggle() {
+        this.setState({
+            showCountries: !this.state.showCountries
+        });
+    }
+
+    render() {
+        const countriesList = this.props.countries.map(c => c.name).join(", ");
+
+        return <span><a href="#" id={this.props.targetKey}
+                       onClick={this.toggle}>view list</a>
+            <Popover placement="right" isOpen={this.state.showCountries} target={this.props.targetKey}
+                     toggle={this.toggle}>
+                <PopoverBody>{countriesList}</PopoverBody>
+            </Popover></span>
+    }
+}
+
+export class ExpectationsDescription extends React.Component<ExpectationMapping, State> {
 
-export class ExpectationsDescription extends React.PureComponent<ExpectationMapping> {
     render(): JSX.Element {
+
+        const expectation = this.props.expectation;
+        const numCountries = expectation.countries.length;
+        const numYears = expectation.years.maximum_inclusive - expectation.years.minimum_inclusive + 1;
+        const numAges = expectation.ages.maximum_inclusive - expectation.ages.minimum_inclusive + 1;
+        const numCohorts = expectation.cohorts.maximum_birth_year - expectation.cohorts.minimum_birth_year + 1;
+        const numOutcomes = expectation.outcomes.length;
+
         return <div className="mt-3 mb-5 p-3 border">
             <div className="h3">Template for {this.applicableScenarios()}</div>
             <div>
-                You need to do the following:
+                Expecting data for:
                 <ul>
-                    <li>These years</li>
-                    <li>These ages</li>
+                    <li id={"years"}>{numYears} years: {rangeAsString(expectation.years)}</li>
+                    <li id={"ages"}>{numAges} ages: {rangeAsString(expectation.ages)}</li>
+                    {expectation.cohorts.minimum_birth_year &&
+                    <li id={"cohorts"}>{numCohorts} cohorts: {expectation.cohorts.minimum_birth_year} - {expectation.cohorts.maximum_birth_year}</li>}
```

Comment:
Talk to Tini, but I think these cohorts need to be phrased with more explanation than just the simple age and year restrictions. Ideally we want this to be clear to prospective modellers without them reading lots of explanations. I was thinking something like these four (based on which values are null):

* "No cohort restrictions"
* "Cohorts: Only people born after 2000"
* "Cohorts: Only people born before 2050"
* "Cohorts: Only people born after 2000 and before 2050"